tests/test_perf_tracking.py

```diff
@@ -1982,15 +1992,112 @@ def test_empty_positions(self):
+        trading.environment.update_asset_finder(asset_metadata=metadata)
+        pt = perf.PositionTracker()
+        dt = pd.Timestamp("1984/03/06 3:00PM")
+        pos1 = perf.Position(1, amount=np.float64(100.0),
+                             last_sale_date=dt, last_sale_price=10)
+        pos2 = perf.Position(2, amount=np.float64(100.0),
+                             last_sale_date=dt, last_sale_price=10)
+        pt.update_positions({1: pos1, 2: pos2})
+
+        event1 = Event({'sid': 1,
+                        'price': 11,
+                        'dt': dt})
+        event2 = Event({'sid': 2,
+                        'price': 11,
+                        'dt': dt})
+
+        # Check cash-adjustment return value
+        self.assertEqual(0, pt.update_last_sale(event1))
+        self.assertEqual(100000, pt.update_last_sale(event2))
+
+    def test_position_values_and_exposures(self):
+        metadata = {1: {'asset_type': 'equity'},
+                    2: {'asset_type': 'equity'},
+                    3: {'asset_type': 'future',
+                        'contract_multiplier': 1000},
+                    4: {'asset_type': 'future',
+                        'contract_multiplier': 1000}}
+        if trading.environment is None:
+            trading.environment = trading.TradingEnvironment()
+        trading.environment.update_asset_finder(asset_metadata=metadata)
+        pt = perf.PositionTracker()
+        dt = pd.Timestamp("1984/03/06 3:00PM")
+        pos1 = perf.Position(1, amount=np.float64(10.0),
+                             last_sale_date=dt, last_sale_price=10)
+        pos2 = perf.Position(2, amount=np.float64(-20.0),
+                             last_sale_date=dt, last_sale_price=10)
+        pos3 = perf.Position(3, amount=np.float64(30.0),
+                             last_sale_date=dt, last_sale_price=10)
+        pos4 = perf.Position(4, amount=np.float64(-40.0),
+                             last_sale_date=dt, last_sale_price=10)
+        pt.update_positions({1: pos1, 2: pos2, 3: pos3, 4: pos4})
+
+        # Test long-only methods
+        self.assertEqual(100, pt._long_value())
+        self.assertEqual(100 + 300000, pt._long_exposure())
+
+        # Test short-only methods
+        self.assertEqual(-200, pt._short_value())
+        self.assertEqual(-200 - 400000, pt._short_exposure())
+
+        # Test gross and net values
+        self.assertEqual(100 + 200, pt._gross_value())
+        self.assertEqual(100 - 200, pt._net_value())
+
+        # Test gross and net exposures
+        self.assertEqual(100 + 200 + 300000 + 400000, pt._gross_exposure())
+        self.assertEqual(100 - 200 + 300000 - 400000, pt._net_exposure())
+
+
+    def test_generate_end_sid_transaction(self):
+        metadata = {1: {'asset_type': 'equity'},
+                    2: {'asset_type': 'future',
+                        'contract_multiplier': 1000}}
+        if trading.environment is None:
```

Comment:
Same comment as above.

This isn't new to your code, but the way we interact with `TradingEnvironment` is all over the place right now.  What we probably should do IMO is remove the module-scope `environment` global and replace it with a class-level attribute, which is accessed only via `TradingEnvironment.instance()`.